lib/src/slices/api_auth_service/model/api_auth_service_account_model.dart

```diff
@@ -1,9 +1,11 @@
+import '../../data_bkg/model/data_bkg_provider_name.dart';
+
 class ApiAuthServiceAccountModel {
   int? accountId;
   String? username;
   String? displayName;
   String? email;
-  String? provider;
+  DataBkgProviderName? provider;
```

Comment:
why do we need a complex object type here? do we have a data provider table?

Reply:
I also don't see the serialization/deserialization for this obj. If you have an account table and provider table, then the model should include the entire provider object and then in serialization/deserialization map it to your row structure.

Reply:
We're loading the provider's from the json, not the DB. We should add a foreign key to this model and use the String representation in the account model.

Reply:
we're going to use the provider name, not an enum